src/ShortestPathFinder1L.java

```diff
@@ -0,0 +1,120 @@
+package components.shortestpathfinder;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+
+import components.map.Map1L;
+import components.queue.Queue1L;
+
+/**
+ * Kernel implementation of ShortestPathFinder component using a map of
+ * locations to queues of edges.
+ *
+ * Convention: - The map contains no null keys. - Each key maps to a queue of
+ * Edge objects with non-null destination names and non-negative weights. - The
+ * graph is undirected: if A connects to B, then B must also connect to A.
+ *
+ * Correspondence: - Each map entry (key, queue of Edge) represents a location
+ * and its direct paths. - The entire map represents an undirected graph where
+ * locations are vertices and Edge objects are undirected edges between them.
+ */
+public final class ShortestPathFinder1L extends ShortestPathFinderSecondary {
+
+    /**
+     * Represents an edge in the graph with a destination and weight.
+     */
+    private static class Edge {
```

Comment:
If you just want an immutable data type to represent edges, consider using Java records. It would cut down on your code here, AND it gives you toString, equals, and hashCode for free.